go-controller/pkg/factory/factory.go

```diff
@@ -461,10 +463,75 @@ func NewWatchFactory(c kubernetes.Interface, stopChan chan struct{}) (*WatchFact
 		return nil, err
 	}
 
+	if err := wf.startup(stopChan); err != nil {
+		return nil, err
+	}
+	return wf, nil
+}
+
+// NewNodeWatchFactory creates only the informers we need for node-level operations,
+// saving bandwidth and memory
+// It only configures informers for:
+// - pods, and only the pods on this node
+// - services
+// - endpoints
+// - nodes
+func NewNodeWatchFactory(nodeName string, c kubernetes.Interface, stopChan chan struct{}) (*WatchFactory, error) {
+	wf := &WatchFactory{
+		iFactory:  informerfactory.NewSharedInformerFactory(c, resyncInterval),
+		informers: make(map[reflect.Type]*informer),
+	}
+	var err error
+
+	// We want to tweak pod lists, so we can't use the shared InformerFactory
+	podSharedInformer := v1coreinformers.NewFilteredPodInformer(
+		c, "", resyncInterval,
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		func(opts *metav1.ListOptions) {
+			opts.FieldSelector = fields.OneTermEqualSelector("spec.nodeName", nodeName).String()
+		})
+
+	wf.informers[podType], err = newInformer(podType, podSharedInformer)
+	if err != nil {
+		return nil, err
+	}
+
+	wf.informers[serviceType], err = newInformer(serviceType, wf.iFactory.Core().V1().Services().Informer())
+	if err != nil {
+		return nil, err
+	}
+	wf.informers[endpointsType], err = newInformer(endpointsType, wf.iFactory.Core().V1().Endpoints().Informer())
+	if err != nil {
+		return nil, err
+	}
+	wf.informers[nodeType], err = newQueuedInformer(nodeType, wf.iFactory.Core().V1().Nodes().Informer(), stopChan)
```

Comment:
does a node care about other nodes? or should we filter here too to just this node?

Reply:
Good point; we can filter here too.